### PR TITLE
Refactor drone shells to use mob spawners

### PR DESCRIPTION
**Pull request by Cenrus**
## About The Pull Request

Drone shells now use mob spawners instead of items, so they will show up in the spawners list.

## Why It's Good For The Game

Maybe ghosts will use the derelict/cogscarab drone spawns now.

## Changelog
:cl:
tweak: Drone shells now appear in the ghost spawners list
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/drone/drones_as_items.dm [hunk 1]
```diff
@@ -6,18 +6,24 @@
 //Drone shells
 
 //DRONE SHELL
-/obj/item/drone_shell
+/obj/effect/mob_spawn/drone/
```
Comment (utilbird) [inline code/modules/mob/living/simple_animal/friendly/drone/drones_as_items.dm]:
```suggestion
/obj/effect/mob_spawn/drone
```

File: code/modules/antagonists/clock_cult/mobs/cogscarab.dm [hunk 2]
```diff
@@ -47,13 +47,18 @@ GLOBAL_LIST_INIT(cogscarabs, list())
...
 	name = "cogscarab construct"
 	desc = "The shell of an ancient construction drone, loyal to Ratvar."
 	icon_state = "drone_clock_hat"
-	drone_type = /mob/living/simple_animal/drone/cogscarab
+	mob_name = "cogscarab"
+	mob_type = /mob/living/simple_animal/drone/cogscarab
+	short_desc = "You are a cogscarab!"
+	flavour_text = {"You are a cogscarab, a tiny building construct of Ratvar. While you're weak and can't leave Reebe, \
+	you have a set of quick tools, as well as a replica fabricator that can create brass for construction. Work with the servants of Ratvar \
+	to construct and maintain defenses at the City of Cogs."}
```
Comment (utilbird) [inline code/modules/antagonists/clock_cult/mobs/cogscarab.dm]:
```suggestion
	flavour_text = "You are a cogscarab, a tiny building construct of Ratvar. While you're weak and can't leave Reebe, \
	you have a set of quick tools, as well as a replica fabricator that can create brass for construction. Work with the servants of Ratvar \
	to construct and maintain defenses at the City of Cogs."
```

File: code/modules/mob/living/simple_animal/friendly/drone/drones_as_items.dm [hunk 3]
```diff
@@ -6,18 +6,24 @@
...
 	layer = BELOW_MOB_LAYER
-
-	var/drone_type = /mob/living/simple_animal/drone //Type of drone that will be spawned
+	density = FALSE
+	death = FALSE
+	roundstart = FALSE
+	short_desc = "You are a drone."
+	flavour_text = {"
+	You are a drone, a tiny insect-like creature. Follow your assigned laws to the best of your ability.
+	"}
```
Comment (utilbird) [inline code/modules/mob/living/simple_animal/friendly/drone/drones_as_items.dm]:
```suggestion
	flavour_text = "You are a drone, a tiny insect-like creature. Follow your assigned laws to the best of your ability."
```

File: code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm [hunk 4]
```diff
@@ -93,10 +96,17 @@
...
 	desc = "A long-forgotten drone shell. It seems kind of... Space Russian."
-	drone_type = /mob/living/simple_animal/drone/derelict
+	icon = 'icons/mob/drone.dmi'
+	icon_state = "drone_maint_hat"
+	mob_name = "derelict drone"
+	mob_type = /mob/living/simple_animal/drone/derelict
+	short_desc = "You are a long forgotten drone!"
+	flavour_text = {"
+	You are a drone on Kosmicheskaya Stantsiya 13. Something has brought you out of hibernation, and the station is in gross disrepair. Build, repair, maintain and improve the station that housed you on activation.
+	"}
```
Comment (utilbird) [inline code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm]:
```suggestion
	flavour_text = "You are a drone on Kosmicheskaya Stantsiya 13. Something has brought you out of hibernation, and the station is in gross disrepair. \
	Build, repair, maintain and improve the station that housed you on activation."
```

**Comment by ItsTisme** [comment]
I thought cogscarabs got axed?

File: code/game/machinery/droneDispenser.dm [hunk 5]
```diff
@@ -30,7 +30,7 @@
 	var/cooldownTime = 1800 //3 minutes
 	var/production_time = 30
 	//The item the dispenser will create
-	var/dispense_type = /obj/item/drone_shell
+	var/dispense_type = /obj/effect/mob_spawn/drone/
```
Comment (PowerfulBacon) [inline code/game/machinery/droneDispenser.dm]:
```suggestion
	var/dispense_type = /obj/effect/mob_spawn/drone
```

**Review by utilbird — APPROVED** [review]

**Pull request merged**